### PR TITLE
some clippy-suggested improvements

### PR DESCRIPTION
**Pull request by llogiq**
This fixes a number of [clippy](https://github.com/Manishearth/rust-clippy) warnings. It's mostly about readability, though a few changes could affect performance (though probably not measurably).

I've left out things to fix later; I thought I'd just push the first batch to see if you like it.

### CODE REVIEW

File: src/cargo/ops/cargo_rustc/layout.rs [hunk 1]
```diff
@@ -76,7 +76,7 @@ impl Layout {
         // the target triple as a Path and then just use the file stem as the
         // component for the directory name.
         if let Some(triple) = triple {
-            path.push(Path::new(triple).file_stem().ok_or(human(format!("target was empty")))?);
+            path.push(Path::new(triple).file_stem().ok_or(human("target was empty".to_string()))?);
```
Comment (killercup) [inline src/cargo/ops/cargo_rustc/layout.rs]: I'm surprised clippy didn't suggest `ok_or_else(|| human(…))`

File: src/cargo/sources/git/utils.rs [hunk 2]
```diff
@@ -610,7 +610,7 @@ pub fn clone(url: &str, target: &Path, config: &Config) -> CargoResult<()> {
                       target.display()))
     })?;
     let refspec = "refs/heads/*:refs/heads/*";
-    fetch(&repo, &url, refspec, &config).chain_error(||{
+    fetch(&repo, url, refspec, &config).chain_error(||{
```
Comment (killercup) [inline src/cargo/sources/git/utils.rs]: bonus points: add blank before `{`

File: src/cargo/ops/cargo_rustc/layout.rs (hunk 1 above)
Comment (llogiq) [inline src/cargo/ops/cargo_rustc/layout.rs]: It did, but I didn't fix all of those.

**Comment by alexcrichton** [comment]
@bors: r+

Thanks!

**Pull request merged**